packages/integrations/vercel/src/lib/redirects.ts

```diff
@@ -66,7 +65,12 @@ function getRedirectLocation(route: RouteData, config: AstroConfig): string {
 	} else if (typeof route.redirect === 'object') {
 		return pathJoin(config.base, route.redirect.destination);
 	} else {
-		return pathJoin(config.base, route.redirect || '');
+		try {
+			const url = new URL(route.redirect!);
+			return url.href;
+		} catch {
```

Comment:
What's the `try` for? The absence of `.redirect` or a parsing error or `URL`?

Reply:
If the argument is not a full URL, the constructor throws, which here means we fallback to the regular processing of the path as internal.

This is outdated, as this PR is based on top of the one you reviewed earlier: #9287. It should not process external links.